stingray/parameterestimation.py

```diff
@@ -0,0 +1,741 @@
+    use_corner=False
+
+
+#### GENERAL IMPORTS ###
+import numpy as np
+import scipy
+import scipy.optimize
+import scipy.stats
+import scipy.signal
+import copy
+
+
+try:
+    from statsmodels.tools.numdiff import approx_hess
+    comp_hessian = True
+except ImportError:
+    comp_hessian = False
+
+
+### own imports
+from . import PSDPosterior
+
+
+class OptimizationResults(object):
+
+
+    def __init__(self, lpost, res, neg=True):
+        """
+        Helper class that will contain the results of the regression.
+        Less fiddly than a dictionary.
+
+        Parameters
+        ----------
+        lpost: instance of Posterior or one of its subclasses
+            The object containing the function that is being optimized
+            in the regression
+
+
+        res: instance of scipy's OptimizeResult class
+            The object containing the results from a optimization run
+
+        """
+        self.neg = neg
+        self.result = res.fun
+        self.p_opt = res.x
+        self.model = lpost.model
+
+
+        self._compute_covariance(lpost, res)
+        self._compute_model(lpost)
+        self._compute_criteria(lpost)
+        self._compute_statistics(lpost)
+
+    def _compute_covariance(self, lpost, res):
+
+        if hasattr(res, "hess_inv"):
+            self.cov = np.asarray(res.hess_inv)
+            self.err = np.sqrt(np.diag(self.cov))
+        else:
+            ### calculate Hessian approximating with finite differences
+            print("Approximating Hessian with finite differences ...")
+            phess = approx_hess(self.p_opt, lpost, neg=self.neg)
+
+            self.cov = np.linalg.inv(phess)
+            self.err = np.sqrt(np.diag(self.cov))
+
+    def _compute_model(self, lpost):
+        self.mfit = lpost.model(lpost.x, *self.p_opt)
+
+
+    def _compute_criteria(self, lpost):
+
+        self.deviance = -2.0*lpost.loglikelihood(self.p_opt, neg=False)
+
+        ## Akaike Information Criterion
+        self.aic = self.result+2.0*self.p_opt.shape[0]
+
+        ### Bayesian Information Criterion
+        self.bic = self.result + self.p_opt.shape[0]*np.log(lpost.x.shape[0])
+
+        ### Deviance Information Criterion
+        ## TODO: Add Deviance Information Criterion
+
+    def _compute_statistics(self, lpost):
+        try:
+            self.mfit
+        except AttributeError:
+            self._compute_model(lpost)
+
+        self.merit = np.sum(((lpost.y-self.mfit)/self.mfit)**2.0)
+        self.dof = lpost.y.shape[0] - float(self.p_opt.shape[0])
+        self.sexp = 2.0*len(lpost.x)*len(self.p_opt)
+        self.ssd = np.sqrt(2.0*self.sexp)
+        self.sobs = np.sum(lpost.y-self.mfit)
+
+    def print_summary(self, lpost):
+
+        print("The best-fit model parameters plus errors are:")
+        for i,(x,y, p) in enumerate(zip(self.p_opt, self.err,
+                                        lpost.model.parnames)):
+            print("%i) Parameter %s: %.5f +/- %.f5"%(i, p, x, y))
```

Comment:
There is a typo in the printing format: It is `%.5f` instead of `%.f5`

Also, I would recommend to use the syntax `"{0:.5f}".format(value)` instead of  `"%.5f" %value`